### ISSUE TITLE
Allow to skip syncing interactions commands and permissions

### ISSUE THREAD

**Issue by l7ssha** [nyxx-discord/nyxx#156]
**Is your feature request related to a problem? Please describe.**

Interactions should allow skipping sync of commands and permissions to avoid unnecessary api requests.

**Describe the solution you'd like**
Implement interface that will decide if syncing should or should not be skiped

**Additional context**
Pseudo code:

```dart
abstract class ICommandsSync {
  FutureOr<bool> shouldSyncCommands(Interable<CommandBuilder> commands); // or whatevet that builder is called
  FutureOr<bool> shouldSyncCommandPermisions(Interable<CommandBuilder> commands);
}

class LockFileCommandsSync implements ICommandsSync {
  // here you impl of lock file
}

class ManualCommandsSync implements ICommandsSync {
  ManualCommandsSync(this.syncCommands, this.syncPermissions);

  FutureOr<bool> shouldSyncCommands() => this.syncCommands;
  FutureOr<bool> shouldSyncCommandPermisions() => this.syncPermissions;
}

Interactions()
  ..syncOnReady(commandsSync: LockFileCommandsSync())
```

**Comment by HarryET** [comment]
Addressed with PR #159

**Comment by l7ssha** [comment]
Maybe it should also support #168?

**Comment by HarryET** [comment]
Yup, could do.

**Comment by HarryET** [comment]
This initial PR has been implemented in #159, And #168 should/might be implemented in #173.
